card/mod/01_core/set/all/location_history.rb

```diff
@@ -0,0 +1,10 @@
+event :discard_deleted_locations, :after=>:store, :on=>:delete do
+  Env[:controller].discard_locations_for self
+  if success.target == self
+    success.target = :previous
+  end
+end
+
+event :save_current_location, :before=>:render_view, :on=>:read do
+  Env[:controller].save_location
+end
```

Comment:
after the event post-it summit and the grand API discussion a relationship therapy for cards and controller?